cdap-watchdog/src/main/java/io/cdap/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -366,17 +368,24 @@ private long processMetrics() {
         }
 
         byte[] currentMessageId = null;
+        boolean gotMessages;
         TopicProcessMeta localTopicProcessMeta =
           new TopicProcessMeta(lastMessageId, Long.MAX_VALUE, Long.MIN_VALUE, 0,
                                TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()),
                                oldestTsMetricName, latestTsMetricName);
         try (CloseableIterator<RawMessage> iterator = fetcher.fetch()) {
+          gotMessages = iterator.hasNext();
           while (iterator.hasNext() && isRunning()) {
```

Comment:
Can we reuse the variable created in the previous line here as we have calculated the value already?

Reply:
No, previous line get variable once (at start of the loop), while here we recheck on each loop iteration if there are more items

Reply:
Oops right